cunumeric/module.py

```diff
@@ -2498,6 +2499,140 @@ def take(
     return a.take(indices=indices, axis=axis, out=out, mode=mode)
 
 
+def _fill_fancy_index_for_along_axis_routines(
+    a_shape: tuple, axis: int, indices: ndarray
```

Comment:
is this specifically `tuple[int, ...]`? If so NdShape (in `.types`) can be used for that